recipe.py

```diff
@@ -35,16 +35,19 @@ def get_ingredients(self):
     def set_ingredients(self,ingredients):
         self._ingredients.append(ingredients)
         if len(set(self._ingredients)) != len(self._ingredients):
-            self._ingredients.remove(ingredients) #I have tested this and it only removes the recently added ingredient
+            self.remove_ingredient(ingredients) #I have tested this and it only removes the recently added ingredient
             return False
     
     def remove_ingredient(self,ingredient):
         self._ingredients.remove(ingredient)
+        #[x for x in self._ingredients if x is not None] #tried to use list comprehension to get rid of the none value in self._ingredients did not succeed
         #Do I need to prevent this removing duplicates if I have already prevented duplication with the set ingredients validation
 
     def display_ingredients(self):
-        print(self.get_ingredients()) #return a pretty string instead
-
+        for ingredient in self._ingredients:
+            print(ingredient) #When i use return it only returns the first value tomato I am not sure why
```

Comment:
With the new ingredient class this will change,

Reply:
Okay so display_ingredients() will still exist but code underneath it will change or is display ingredients not needed?

Reply:
Yep, ultimately it's the UI that will display to the user, but pretty printing should be added to classes to help with debug and test. There is a way of baking in "sensible" printing of class contents by default (so you can simplify do print(myClassInstance) rather than print(myClassInstance.fancy_print()) ) but it doesn't make sense in all cases and can be misleading at this level.

Reply:
ok, I will not use it in for this class